src/Adapter/Product/Repository/ProductMultiShopRepository.php

```diff
@@ -336,6 +336,90 @@ public function getAssociatedShopIds(ProductId $productId): array
         return $shops;
     }
 
+    /**
+     * @param ProductId $productId
+     *
+     * @return bool
+     */
+    public function hasCombinations(ProductId $productId): bool
+    {
+        $result = $this->connection->createQueryBuilder()
+            ->select('pa.id_product_attribute')
+            ->from($this->dbPrefix . 'product_attribute', 'pa')
+            ->where('pa.id_product = :productId')
+            ->setParameter('productId', $productId->getValue())
+            ->execute()
+            ->fetchOne()
+        ;
+
+        return !empty($result);
+    }
+
+    /**
+     * @param ProductId $productId
+     */
+    public function updateCachedDefaultCombination(ProductId $productId): void
```

Comment:
if its plural, it will be confusing, because it suppose to only update the attribute of cached_default_attribute in a product.
Ill see if i can add shopConstraint, but I don't remember now, maybe the whole idea of this method was to unify the default attribute according to the shops (kinda to clean up the state), so it would loose the whole purpose if i add shop constraint. Let me see.

Reply:
ok added shop constraint, seems to work alright